fw/cache.c

```diff
@@ -328,21 +328,52 @@ tfw_cache_key_node(unsigned long key)
 	return key % num_online_nodes();
 }
 
+/**
+ * Release memory dynamically allocated
+ * for every node cpus
+ */
+static void
+tfw_release_node_cpus(void)
+{
+	int i;
+
+	for (i = 0; i < MAX_NUMNODES; i++)
+	{
+		if(!c_nodes[i].cpu) {
+			T_ERR("%s c_node deallocation interrupted on node %i\n", __func__, i);
+			break;
+		}
+		kfree(c_nodes[i].cpu);
+	}
+}
+
 /**
  * Just choose any CPU for each node to use queue_work_on() for
  * nodes scheduling. Reserve 0th CPU for other tasks.
  */
 static void
 tfw_init_node_cpus(void)
```

Comment:
This function should return int. And in case if allocation failed you should return -ENOMEM; I catch kernel panic when c_nodes[i].cpu = kmalloc return NULL, because  `tfw_release_node_cpus` was called twice.